src/components/selection-list/selection-list.tsx

```diff
@@ -84,92 +96,166 @@ export interface SelectionListProps extends OptionList {
     className?: string;
 }
 
-@SBComponent(style)
-export class SelectionList extends React.Component<SelectionListProps, {}> {
+export interface SelectionListState {
+    focused: boolean;
+    focusedValue?: string;
+}
+
+@SBComponent(listStyle)
+export class SelectionList extends React.Component<SelectionListProps, SelectionListState> {
     public static defaultProps: SelectionListProps = {
         dataSource: [],
-        itemRenderer: DefaultItemRenderer,
+        renderItem,
         onChange: () => {}
     };
 
+    public state: SelectionListState = {
+        focused: false,
+        focusedValue: undefined
+    };
+
+    private focusableItemValues: string[] = [];
+
+    public componentWillReceiveProps(nextProps: SelectionListProps) {
+        if (this.state.focused) {
+            this.setState({focusedValue: nextProps.value});
+        }
+    }
+
     public render() {
         const rootProps = root(this.props, {
             'data-automation-id': 'LIST',
             'className': 'list',
-            'onClick': this.handleClick
+            'cssStates': {
+                focused: this.state.focused
+            },
+            'tabIndex': -1,
```

Comment:
`tabIndex` should be from props.
`on-*` behavior should be examined, if not overriden by user - put them on `root`